Calculate and store metrics for the specified k

epugh opened the pull request:
### Description
* Enhance `demo.sh` script to calculate implicit judgements from UBI and then run a POINTWISE experiment.
* You can pass `--skip-ecommerce-step` to script to not delete and reload the `ecommerce` index.

### Issues Resolved
Closes #25 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

epugh commented:
This PR incorporates the work done in #37, so will close #37 in favour of this one.

wrigleyDan commented on src/test/scripts/demo.sh:
```diff
@@ -131,15 +250,38 @@ exe curl -s -X PUT "localhost:9200/_plugins/search_relevance/experiments" \
+
+echo
+echo Create POINTWISE Experiment
+# TODO the type UBI_EVALUTATION SHOULD BE POINTWISE_EVALUATION
```
This TODO is no longer necessary, right?

wrigleyDan commented on CHANGELOG.md:
```diff
@@ -21,4 +21,5 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
  - LLM Judgment Processor Improvement ([#27](https://github.com/opensearch-project/search-relevance/pull/27))
  - Deal with experiment processing when no experiment variants exist. ([#45](https://github.com/opensearch-project/search-relevance/pull/45))
 
+
```
I guess a line what was changed in the demo script would be nice here instead of an empty line :)

wrigleyDan reviewed (commented):
I wasn't able to see a successful POINTWISE_EXPERIMENT.

I guess that might be related to other discussions from the last days. I'd love to see the whole script work and produce meaningful results before merging it to validate that there are no blocking issues left in the basic functionality.

epugh commented:
> Thanks for addressing the feedback.
> 
> Technically, I have no objections. With the reduced data being indexed and the judgment list of all 0s that's used for the experiments we stripped out meaningful experiment results. If a technical demonstration is our goal for this script this works absolutely fine. Given the time constraints I'd probably go for that direction.
> 
> I suggest creating a separate script for the hybrid search experiment including model registration and all that's required for a "real hybrid search scenario". That should be done in another issue.

Just to get this one to done, I split out from `demo.sh` into `demo_hybrid_optimizer.sh` the HO stuff.   

I have a demo to marketing folks later today, so hopefully we can get `demo_hybrid_optimizer.sh` updated..   This at least works.  I will also try and enhance the demo.

If this change looks good enough, then let's get this merged since it has a fix for search eval calculations and is getting rather long!
